File: recipes/java-recipes/spring-boot-postgres-recipe/src/main/java/com/trendyol/stove/examples/java/spring/infra/boilerplate/kafka/TopicResolver.java
package com.trendyol.stove.examples.java.spring.infra.boilerplate.kafka;

public class TopicResolver {
    private final KafkaConfiguration kafkaConfiguration;

    public TopicResolver(KafkaConfiguration kafkaConfiguration) {
        this.kafkaConfiguration = kafkaConfiguration;
    }

    public Topic resolve(String aggregateName) {
        return kafkaConfiguration.getTopics().get(aggregateName);
    }
}

File: recipes/jvm/kotlin-recipes/ktor-postgres-recipe/src/main/kotlin/com/trendyol/stove/examples/kotlin/ktor/infra/boilerplate/kafka/TopicResolver.kt
package com.trendyol.stove.examples.kotlin.ktor.infra.boilerplate.kafka

import com.trendyol.stove.examples.kotlin.ktor.application.*

class TopicResolver(
  private val kafkaConfiguration: KafkaConfiguration
) {
  operator fun invoke(aggregateName: String): Topic = kafkaConfiguration.topics.getValue(aggregateName)
}
